src/app/components/create-dataset-modal/create-dataset-modal.component.ts

```diff
@@ -133,9 +134,15 @@ export class CreateDatasetModal {
   @ViewChild('dsorg') dsorg: ElementRef;
 
 
-  constructor(private el: ElementRef,) { }
-
-  ngOnInit() {
+  constructor(private el: ElementRef, @Inject(MAT_DIALOG_DATA) data ) {
+    if(data && data.data) {
+      this.properties.name = data.data.dsName;
+      this.properties.datasetNameType = data.data.dsNameType;
+      this.properties.organization = data.data.dsOrg;
+    } else {
+      this.properties.datasetNameType = 'PDS';
+      this.properties.organization = 'PO'
+    }
```

Comment:
Can you explain this part? It looks like you're enhancing create dataset menu to be prefilled with stuff which is excellent but why are we assuming those last two values? Does it make sense? Maybe it does, don't know too much about DS but just checking... I'm guessing those are prefilled values that appear in the menu and the user can change them if they want to right

Reply:
Yes. The dataset name and organization are prefilled in the ds creation form. And it can be changed from the drop-down menu.